packages/wonder-blocks-form/src/components/labeled-text-field.js

```diff
@@ -4,16 +4,68 @@ import * as React from "react";
 import FieldHeading from "./field-heading.js";
 import TextField from "./text-field.js";
 
-export default class LabeledTextField extends React.Component<{||}> {
+type Props = {|
+    /**
+     * Makes a read-only input field that cannot be focused. Defaults to false.
+     */
+    disabled: boolean,
+|};
+
+type DefaultProps = {|
+    disabled: $PropertyType<Props, "disabled">,
+|};
+
+type State = {|
+    /**
+     * Displayed when the validation fails.
+     */
+    error: ?string,
+
+    /**
+     * The user focuses on the textfield.
+     */
+    focused: boolean,
+|};
+
+export default class LabeledTextField extends React.Component<Props, State> {
+    static defaultProps: DefaultProps = {
+        disabled: false,
+    };
+
+    state: State = {
+        error: null,
+        focused: false,
+    };
+
+    handleOnFocus: (
+        event: SyntheticFocusEvent<HTMLInputElement>,
+    ) => mixed = () => {
+        this.setState({
+            focused: true,
+        });
+    };
+
+    handleOnBlur: (
+        event: SyntheticFocusEvent<HTMLInputElement>,
+    ) => mixed = () => {
+        this.setState({
+            focused: false,
+        });
```

Comment:
Where is `this.state.focused` used?

Reply:
It will be used later when I add validation. I will only be showing the error message (if any) when the user is **not** focused on the textfield.